### ISSUE TITLE
[ Installation Error ] Met 'exit code -9' error after executing the `wasp start `

### ISSUE THREAD

**Issue by godblessbb** [wasp-lang/open-saas#20]
I'm trying to run this project on my CentOS7 server, and since the system version is not compatible with some of the project's dependencies I'm using docker to run the project and the database.

Errors are consistently encountered while executing the `wasp start `command:

- npm install failed: Setup failed! Server setup failed with exit code -9. Web app setup failed with exit code -9.

![WX20240121-224855@2x](https://github.com/wasp-lang/open-saas/assets/4539598/1c0d3f9e-de8c-4939-ad0f-f2f0c1081d91)

My consultation with ChatGPT resulted in saying that my system is low on memory, I have only 1G of RAM and only about 500M free, but I'm not convinced that this is the cause of the problem as I've manually installed all the dependencies to minimize the memory usage of wasp, but the problem is still not solved. Can anyone help me?

**Comment by vincanger** [comment]
Hm. Thanks for making us aware of this. Perhaps @Martinsos or @infomiho have some insight into this issue?

**Comment by Martinsos** [comment]
@godblessbb thanks for info, I am not sure at the moment what the cause could be!

Which Docker image are you using, can you share the Dockerfile?

exit code -9 is quite weird, quick googling tells me it might be related to sigkill -> but that sounds weird. Did you kill the process to stop it, or did these errors happen on their own?

Only 1GB of RAM and only 500M free -> that might be the problem, I don't think that is enough memory.

**Comment by godblessbb** [comment]
@Martinsos Hi Martin, Thank you for your reply.

I'm a beginner in programming and I'm trying to build your program with the help of ChatGPT, I thought I could run through it with a few lines of code, but it's actually probably more difficult than I thought it would be.

Let me elaborate on my current situation. I've been trying to wrestle with this project for a few days now. 

1. At first I was using a CentOS7 server, I was using one docker container to run the project files and another docker container to run the database and establish communication between them, in this case I got the code to `wasp start `and then ran into a fatal bus error, after a few days of fruitless attempts I chose to give up.

<img width="770" alt="image" src="https://github.com/wasp-lang/open-saas/assets/4539598/3c15d16a-a689-4c46-86c2-618c4bcc0510">

2. As of yesterday, I purchased a new server. It's a brand new server with CentOS 9. It has 1 core and 1g of RAM like the previous server, it's not a very high configuration, but considering I'm only using it as a test, I thought it would be enough and this new server will have only opensaas for one project. This time I tried to follow the tutorial exactly and just run the database with docker. But this time I got as far as `wasp db migrate-dev `and couldn't proceed.

<img width="843" alt="image" src="https://github.com/wasp-lang/open-saas/assets/4539598/cbc4915e-4ce8-48da-baf8-4686aaa87ecd">

This is what I'm experiencing right now. A few things to clarify for you are  
1️⃣ While the machine is not highly configured, I suspect that this may not be the main cause, especially now that I'm using a brand new server and it's still giving me errors.  
2️⃣ I've tried installing a few dependencies manually, and that moved my installation forward a bit until I ran into the fatal bus error.  
3️⃣ In my attempts today, I've been stuck in the @ swc/core installation, I was told that the process was accessing illegal or non-existent memory. It also looks like a bus error.  
4️⃣ I suspect it's not my dockerfile causing the problem, as I didn't use dockerfile to create the project containers on the new server, but I still pasted the previous dockerfile below.  
5️⃣ Is there some sort of unknown system conflict? I even tried installing the dependencies manually via package.json, but I still get the bus error.

I'm really looking forward to using this program so I could really use your help and would be happy to help you find any problems you haven't encountered. I hope to hear from you. dockerfile See below:


> # 使用 Rust 官方镜像作为基础
> FROM rust:latest
> 
> # 安装 curl 和 Node.js
> RUN apt-get update && apt-get install -y curl
> RUN curl -fsSL https://deb.nodesource.com/setup_18.x | bash -
> RUN apt-get install -y nodejs
> 
> # 安装 Wasp
> RUN curl -sSL https://get.wasp-lang.dev/installer.sh | sh
> 
> # 将 Wasp 安装路径添加到 PATH
> ENV PATH="/root/.local/bin:${PATH}"
> 
> # 尝试运行 wasp 命令确认安装成功
> RUN wasp --version
> 
> 
> # 设置工作目录
> WORKDIR /app
> 
> # 将当前目录内容复制到位于/app中的容器中
> # 这里假设您的 Wasp 项目文件都在您的 Dockerfile 目录中
> COPY . /app
> 
> # 暴露所需端口（如果您的应用需要监听端口）
> # 例如，如果 Wasp 服务监听 3000 端口
> EXPOSE 3000
> 
> # 容器启动时运行 Wasp
> # 请根据您的实际需求替换以下命令
> CMD ["wasp", "start"]

**Comment by godblessbb** [comment]
@vincanger @Martinsos @infomiho  I suspect that your project is conflicting with CentOS, can you tell me if you are able to run it correctly on CentOS? Also, what is the minimum configuration to run this project, I don't really believe that it is the configuration that is causing the compilation to fail.

**Comment by godblessbb** [comment]
It seems I solved the problem by manually removing node_modules and manually npm installing package.json. this problem is now solved, now working on the new one.

**Comment by Martinsos** [comment]
@godblessbb sorry for late answer, we are releasing Wasp 0.12 this week and have been a bit overwhelmed so this issue has slipped our hands for a bit. I know you said in another comment you are giving up with running Wasp on CentOS for now but I will still respond here, to not leave the issue unanswered.

It is a bit hardto say for me what is causing the error exactly. I am suspecting that the fixes you made manually did not really fix the issue and that is why you are just getting other errors.
You said you ran it on CentOS9 now, but had errors during npm install. Without more logs, it is really hard to say what is the issue. I don't currently have CentOS9 installation i can easily test this on. Have you maybe checked for those specific libraries, if they have any issues with CentOS9 reported?
One certain way to get this resolved is to run Wasp on Ubuntu, since it is built on Ubuntu.

As for your dockerfile you used previously -> it is a bit weird that it is Rust based, I think you would probably instead want to use `node` based one.

**Comment by godblessbb** [comment]
@Martinsos Thank you for your response, Martin. I am currently specifying the system on the Dockerfile as Ubuntu and continuing with my attempts. So far, there doesn't seem to be any significant improvement, but I still want to give it a try. I sincerely suggest that you consider creating a Docker image for the installation process. This would make the installation faster and more user-friendly for those with lower technical skills, as many people who choose Wasp may not be proficient in frontend development. Thank you again, Martin.

**Comment by godblessbb** [comment]
@Martinsos Hi Martin, http://172.105.127.24:3000/
It looks like I have successfully set it up. Can you help me click on "Pricing" on the homepage? It takes me several seconds to open it here. Is it the same for you?

**Comment by Martinsos** [comment]
At current stage, Wasp is actually meant for devs who have some frontend experience (React, HTML, ...) -> that is why we are not yet prioritizing polishing all the rough edges, but will in the future. It is therefore valuable for us to get feedback, although we might not be able to prioritize acting on it always.

Nice job on getting Wasp running! I tried the link, and pricing page takes long time indeed, as does initial load also. For some reason, your app seems to be quite slow (Not just pricing page, but any page that is served by your Wasp app). It is hard for me to say what the reason is, might be the setup you have, or machine doesn't have enough resources, or something like that.

Docker image for the installation process -> ideally you would install Wasp on your machine by using our installer, and run it directly on your machine. While Wasp can be installed inside Docker and then run inside it, it is not an experience we are aiming to specifically support at the moment as it has its own complications. I wish we did, but we just don't have the manpower at the moment, we are a small team trying to do a lot at the same time and just can't cover everything :), so my apologies for not being able to support you better on this.
